lib/Doctrine/ORM/Proxy/ProxyFactory.php

```diff
@@ -102,23 +164,38 @@ public function __serialize(): array
      */
     public function __construct(EntityManagerInterface $em, $proxyDir, $proxyNs, $autoGenerate = self::AUTOGENERATE_NEVER)
     {
-        $proxyGenerator = new ProxyGenerator($proxyDir, $proxyNs);
-
-        if ($em->getConfiguration()->isLazyGhostObjectEnabled()) {
-            $proxyGenerator->setPlaceholder('baseProxyInterface', Proxy::class);
-            $proxyGenerator->setPlaceholder('useLazyGhostTrait', Closure::fromCallable([$this, 'generateUseLazyGhostTrait']));
-            $proxyGenerator->setPlaceholder('skippedProperties', Closure::fromCallable([$this, 'generateSkippedProperties']));
-            $proxyGenerator->setPlaceholder('serializeImpl', Closure::fromCallable([$this, 'generateSerializeImpl']));
-            $proxyGenerator->setProxyClassTemplate(self::PROXY_CLASS_TEMPLATE);
-        } else {
+        if (! $em->getConfiguration()->isLazyGhostObjectEnabled()) {
```

Comment:
should we trigger this deprecation only on PHP 8.1+ ? When using older PHP versions, they cannot enable it as they cannot install the needed dependency.

Reply:
We can but this won't save ppl from this one:
https://github.com/doctrine/common/pull/1002/files#diff-58c63ccd37f4120e07f5995493641146490fc19779d4865aa4669fc1ce6282ef

I have a slight preference to trigger the deprecation regardless of the PHP version. Please advise.

Reply:
Well, this one should probably also be conditional, or the ORM should bump its PHP min version first.

Reply:
I added the version check.